modules/lldp/cli.c

```diff
@@ -0,0 +1,209 @@
+	void *resp_ptr;
+	(void)p;
+
+	if (gr_api_client_send_recv(c, GR_LLDP_SHOW_CONFIG, 0, NULL, &resp_ptr) < 0)
+		return CMD_ERROR;
+
+	resp = resp_ptr;
+	printf("lldp: rx: %s tx: %s ttl: %d\n",
+	       resp->rx ? "on" : "off",
+	       resp->tx ? "on" : "off",
+	       resp->ttl);
+
+	return CMD_SUCCESS;
+}
+
+static void decode_lldp(uint16_t data_len, uint8_t *data, char *ifname, uint8_t brief) {
+	uint16_t offset = 0;
+	char port_desc[512];
+	char sys_name[512];
+	char sys_desc[512];
+	uint16_t ttl = 0;
+
+	do {
+		uint8_t type = data[offset + 0] >> 1;
+		uint16_t len = ((data[offset + 0] & 0x1) << 8) + data[offset + 1];
+		offset += 2;
+		switch (type) {
+		case T_CHASSIS_ID:
+			break;
+		case T_PORT_ID:
+			break;
+		case T_TTL:
+			ttl = ntohs(*(uint16_t *)&data[offset]);
+			break;
+		case T_PORT_DESC:
+			memcpy(port_desc, &data[offset], len);
+			port_desc[len] = '\0';
+			break;
+		case T_SYSTEM_NAME:
+			memcpy(sys_name, &data[offset], len);
+			port_desc[len] = '\0';
+			break;
+		case T_SYSTEM_DESC:
+			memcpy(sys_desc, &data[offset], len);
+			port_desc[len] = '\0';
+			break;
+		case T_END:
+			goto end;
+			break;
+		default:
+			break;
+		}
+end:
+		offset += len;
+	} while (offset < data_len);
+
+	printf("%3ds   %s \t%s \t", ttl, ifname, sys_name);
+	if (brief)
+		printf("\n");
+	else
+		printf("%s \t%s \t%s\n", sys_name, sys_desc, port_desc);
+}
+
+static cmd_status_t lldp_show_neighbors(const struct gr_api_client *c, const struct ec_pnode *p) {
+	struct gr_lldp_show_neighbors_resp *resp = NULL;
+	struct gr_iface iface;
+	uint8_t brief = 1;
+	void *resp_ptr;
+	(void)p;
+
+	if (gr_api_client_send_recv(c, GR_LLDP_SHOW_NEIGH, 0, NULL, &resp_ptr) < 0)
+		return CMD_ERROR;
+
+	resp = resp_ptr;
+	printf("LLDP Neighbors: %d\n", resp->n_neigh);
+	printf("Age  TTL  ifname\tSys name\t");
+	if (brief)
+		printf("\n");
+	else
+		printf("sys desc\tport_desc\n");
+	for (uint16_t i = 0; i < resp->n_neigh; i++) {
+		clock_t age = (resp->now - resp->neighbors[i].last_seen) / 1000000;
+
+		if (iface_from_id(c, resp->neighbors[i].iface_id, &iface) < 0)
+			return CMD_ERROR;
+
+		printf("%3ld ", age);
+		decode_lldp(
+			resp->neighbors[i].n_tlv_data,
+			resp->neighbors[i].tlv_data,
+			iface.name,
+			brief
+		);
+	}
+
+	return CMD_SUCCESS;
+}
+
+static int ctx_init(struct ec_node *root) {
+	int ret;
+
+	ret = CLI_COMMAND(
+		CLI_CONTEXT(root, CTX_SET, CTX_ARG("lldp", "Modify LLDP configuration")),
+		"(rx|tx|both|off)",
+		lldp_enable,
+		"Enable globally lldp",
+		with_help("Enable RX Only", ec_node_str("rx", "rx")),
+		with_help("Enable TX Only", ec_node_str("tx", "tx")),
+		with_help("Enable TX and RX", ec_node_str("both", "both")),
+		with_help("Disable", ec_node_str("off", "off"))
+	);
+
+	ret = CLI_COMMAND(
+		CLI_CONTEXT(root, CTX_SET, CTX_ARG("lldp", "Modify LLDP configuration")),
+		"ttl TTL",
+		lldp_set_ttl,
+		"Set LLDP TTL",
+		with_help("Interval in seconds, 10 < ttl < 600", ec_node_uint("TTL", 10, 600, 10))
+	);
```

Comment:
Can you merge these two commands into a single one and use a set_attrs field in the API message to determine which settings are to be considered?

https://github.com/rjarry/grout/blob/bed15f4ef48ebf770d84fbb9f52be736c742ac06/modules/infra/api/gr_infra.h#L142

Reply:
Thanks ! That's what I was looking for :)